app/src/main/java/ar/com/wolox/android/example/ui/news/NewsPresenter.kt

```diff
@@ -36,4 +42,46 @@ class NewsPresenter @Inject constructor(
             }
         })
     }
+
+    fun toggleLike(post: News?) {
+        val currentUserId = sharedPref.getInt(mPrefUserId, 0)
+        var like = false
+        if (post?.likes?.contains(currentUserId) == true) {
+            (post.likes as ArrayList).remove(currentUserId)
+        } else {
+            (post?.likes as ArrayList).add(currentUserId)
+            like = true
+        }
+
+        callToggleLike(post.likes, post.id, like)
+    }
+
+    private fun callToggleLike(likes: List<Int>?, newsId: Int, like: Boolean) {
+
+        val call = newsId.let { retrofitServices.getService(NewsService::class.java).toogleLike(NewsLikes(likes), it) }
```

Comment:
Well, this not ok. If news id is non-null, it is not necessary to use `newsId.let {}`

Reply:
Moreover, it's not necessary to save the call and then do the enqueue, do it directly. (I believe that I'm not being clear, so ask me if you don't get it)